### PR TITLE
Ask for authorization overrides

### PR DESCRIPTION
**Pull request by erikwiffin**
Fixes https://github.com/sync-for-science/tracking/issues/11#issuecomment-229064968.

### CODE REVIEW

**Comment by jmandel** [comment]
For the "same code twice" issue, I'm expecting to see a `code` value being overridden, with a previously-supplied value re-supplied. Is this commit going through that workflow?

**Comment by erikwiffin** [comment]
Oh, missed that line for some reason. Should be easy to add.

**Comment by erikwiffin** [comment]
And this uncovered a bug in the `auth-proxy` codebase where we aren't properly expiring grants after they've been used. Looks like the expiration is being set, but the SQLAlchemy transaction isn't getting committed.

File: features/20-exchange-access-token.feature [hunk 1]
```diff
@@ -58,3 +58,11 @@ Feature: App exchanges authorization code for access token
             | key | value |
             | redirect_uri | https://example.com |
         Then the response code should not be 200
+
+    Scenario: Use received code twice
+        Given OAuth is enabled
+        And I am not logged in
+        When I ask for authorization
+        And I exchange my authorization code
```
Comment (jmandel) [inline features/20-exchange-access-token.feature]: I love that this is the test :)

**Pull request merged**